src/adapters/scene.js

```diff
@@ -399,4 +407,30 @@ Scene.prototype.moveMobileGeolocationButton = function(bottom = 0) {
   }
 };
 
+Scene.prototype.mobileGeolocationButtonVisibility = function(visible) {
+  if (isMobileDevice()) {
+    const item = document.querySelector('.mapboxgl-ctrl-geolocate');
+    if (item) {
+      if (visible) {
+        item.classList.remove('hidden');
+      } else {
+        item.classList.add('hidden');
+      }
+    }
+  }
+};
+
+Scene.prototype.mobileDirectionButtonVisibility = function(visible) {
```

Comment:
`mobileDirectionButtonVisibility` and `mobileGeolocationButtonVisibility` are nearly identical, can we define and reuse a more generic function like `setMobileDirectionButtonVisibility(selector, isVisible)` ?

Reply:
sure, but in the function only, the 4 events remain the same to avoid copying the css selector in every "fire" :)
thanks